jinja_unquote_resolvers_filter/unquote_resolvers.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any, Dict, List, Union
+
+import yaml
+from yaml.nodes import ScalarNode
+
+from functools import singledispatch
+from textwrap import indent as textwrap_indent
+
+try:
+    from yaml import CSafeDumper as SafeDumper
+except ImportError:
+    from yaml import SafeDumper  # type: ignore
+
+
+class CustomDumper(SafeDumper):
+    pass
+
+
+def unquoted_tag_representer(dumper: CustomDumper, data: "Tag") -> ScalarNode:
+    return dumper.represent_scalar(data.tag, data.value, style="")
+
+
+class Tag:
+    def __init__(self, tag: str, value: str) -> None:
+        self.tag = tag
+        self.value = value
+
+
+CustomDumper.add_representer(Tag, unquoted_tag_representer)
+
+
+def unquote_resolvers(
+    data: Union[Dict, List],
+    indent: int = 2,
+    output_indent: int = 0,
+    trim: bool = False,
+    *args: Any,
+    **kw: Any
+) -> str:
+    """
+    Remove quotes around resolver expressions and
+    produce nice YAML.
+
+    :param data: The input data as a dictionary or list
+      containing resolver expressions.
+    :param indent: The number of spaces to use for indentation
+      of nested structures in the output YAML.  Default is 2
+      spaces.
+    :param output_indent: The number of spaces to use for
+      indentation of the entire output YAML. Default is 0
+      (no indentation).
+
+    :returns: A formatted YAML string with unquoted resolver
+      expressions and the specified indentation settings.
+    """
+
+    @singledispatch
+    def process_item(item: Any) -> Any:
+        return item
+
+    @process_item.register(str)
+    def _(item: str) -> Any:
+        if item.startswith("!"):
+            if "\n" in item:
+                raise NotImplementedError("Multiline expressions not supported")
```

Comment:
Nice. Thanks for the clarification. I have no idea what would have happened with it being multiline. This is a great way to draw a nice boundary between supported and not supported.